{{cookiecutter.package_name}}/src/{{cookiecutter.package_name}}/tools/cli.py

```diff
@@ -6,136 +6,159 @@
     """
-    def helper(item):
-        valid, loc = sanitize(item)
-        if valid:
-            locs.append(loc)
-
-    locs = []
-    if single_loc:
-        helper(single_loc)
-    else:
-        with loc_file.open('r') as infile:
-            for line in infile:
-                helper(line)
-    return locs
-
-
-def log_and_exit(f: Callable) -> Callable:
-    @wraps(f)
-    def wrapper(*args, **kwds):
-        f(*args, **kwds)
-        exit(1)
-    return wrapper
-
-    
-def verify_args(template: Path, locations_file: Path, single_location: str, output_dir: Path) -> None:
-    exit_logger = log_and_exit(logger.error)
-    if not template.exists():
-        exit_logger(f'\nError: the template file "{template}" does not exist.')
-    if not locations_file.exists():
-        exit_logger(f'\nError: the locations file "{locations_file}" does not exist. Create locations.txt '
-                    'or use the "-s" switch to specify a single location.')
-    if not single_location and not locations_file.stat().st_size:
-        exit_logger(f'\nError: the locations file "{locations_file}" is empty. Add locations '
-                    'to this file or use the "-s" switch to specify a single location.')
-    if not output_dir.exists():
-        exit_logger(f'\nError: the output directory "{output_dir}" does not exist.', exit=True)
-
-
-MODEL_SPEC_DIR = Path(__file__).parent.parent / 'model_specifications'
+    if locations_file and single_location:
+        raise ValueError('You provided both a locations file and a single location to make_specs.')
+
+    if single_location:
+        return [single_location]
+
+    locations_file = Path(locations_file) if locations_file else MODEL_SPEC_DIR / 'locations.txt'
+    with locations_file.open() as f:
+        # Interpret each line that doesn't start with a '#' as a single location.
+        locations = [l for l in f.readlines() if not l.startswith('#')]
+    if not locations:
+        raise ValueError(f'No locations provided in location file {str(locations_file)}.')
+
+    return locations
+
+
+MODEL_SPEC_DIR = (Path(__file__).parent.parent / 'model_specifications').resolve()
 @click.command()
-@click.option('-l', '--locations_file',
-              default=MODEL_SPEC_DIR / 'locations.txt',
-              show_default=True,
+@click.option('-l', '--locations-file',
               type=click.Path(dir_okay=False),
-              help=f'The file with the location parameters for the template.')
+              help=('The file with the location parameters for the template. If no locations file is provided '
+                    'and no single location is provided with the ``-s`` flag, this will default to the locations '
+                    f'file located at {str(MODEL_SPEC_DIR / "locations.txt")}.'))
 @click.option('-t', '--template',
-              default=MODEL_SPEC_DIR / 'model_spec.in',
+              default=str(MODEL_SPEC_DIR / 'model_spec.in'),
```

Comment:
Pathlib paths don't display as a user probably expects in a help string.  I try to always coerce paths to strs in help messages and errors because the user shouldn't know or care about the internal representation.